### ISSUE TITLE
Future-proof the rIC API and make it extensible

### ISSUE THREAD

**Issue by igrigorik** [w3c/requestidlecallback#21]
```
requestIdleCallback(IdleRequestCallback callback, optional unsigned long timeout);
```

**Problem:** Current API signature takes an optional timeout value, which makes it impossible to extend this API in the future with other optional or required parameters. 

**Proposal:** move timeout into an optional dictionary…

```
requestIdleCallback(IdleRequestCallback callback, optional IdleRequestOptions options);

dictionary IdleRequestOptions {
             optional unsigned long timeout;
};
```

**Comment by rmcilroy** [comment]
SGTM. @toddreifsteck, any objections?

**Comment by bzbarsky** [comment]
If you really wanted to, you could even support both:

```
requestIdleCallback(IdleRequestCallback callback, unsigned long timeout);
requestIdleCallback(IdleRequestCallback callback, optional IdleRequestOptions options);
```

No idea whether you want that.  Just having the dictionary is probably fine.

**Comment by igrigorik** [comment]
FWIW, given that we're working with clean slate here, I'd vote for dictionary only -- reduces confusion.

**Comment by eliperelman** [comment]
@igrigorik no objections. I'm also fine with dictionary-only.

**Comment by rmcilroy** [comment]
Change uploaded to address this in #22

**Comment by toddreifsteck** [comment]
Perfect! LGTM

**Comment by igrigorik** [comment]
Resolved via #22, closing.
